### PR TITLE
add IO functionalities to write DCIP2D mesh and models

### PR DESCRIPTION
**Pull request by thast**
- the `MeshIO` functions `writeUBC` and `writeModelUBC` are now compatible with 2D meshes and models.

- We now make sure that mesh and model get saved at the same place by giving the option to specify the folder outside of the filename (before the folder was given by the `dict.keys`, not very practical)

- The upgrade is backward compatible

### CODE REVIEW

File: discretize/MeshIO.py [hunk 1]
```diff
@@ -300,7 +300,56 @@ def _toVTRObj(mesh, models=None):
             vtkObj.GetCellData().SetActiveScalars(models.keys()[0])
         return vtkObj
 
-    def readModelUBC(mesh, fileName):
+    def _readModelUBC_2D(mesh, fileName):
+        """
```
Comment (lheagy) [inline discretize/MeshIO.py]: Thanks for this @thast!

File: discretize/MeshIO.py [hunk 2]
```diff
@@ -300,7 +300,56 @@ def _toVTRObj(mesh, models=None):
...
+
+            for ii in range(len(obsfile)-1):
+                mm = np.array(obsfile[ii+1].split(), dtype=float)
+                model[:, ii] = mm
+
+            model = model[:, ::-1]
+
+        else:
+
+            if len(obsfile[1:]) == 1:
```
Comment (lheagy) [inline discretize/MeshIO.py]: would you mind adding a test that hits these lines?

File: tests/base/2d_2cyl_model [hunk 3]
```diff
@@ -0,0 +1,48 @@
+135 47
```
Comment (lheagy) [inline tests/base/2d_2cyl_model]: @thast: is this for testing, if you slack me the file, I will upload and send you a url where you can download it so it is not in the core repo

File: tests/base/ubc_DC2D_tensor_mesh.msh [hunk 4]
```diff
@@ -1,35 +1,26 @@
...
-2700 1
-2750 1
-2800 1
-2900 1
-3050 1
-3250 1
-3550 1
-3950 1
-4550 1
+!comment line
```
Comment (lheagy) [inline tests/base/ubc_DC2D_tensor_mesh.msh]: same here, I am happy to upload this

File: tests/base/2d_2cyl_model (hunk 3 above)
Comment (thast) [reply tests/base/2d_2cyl_model]: @lheagy : Thanks for pointing it out. The test actually writes and then read back the testing files here (so we test both input and output). 
I will reproduce what we do in SimPEG and create a temporary folder that we then delete, so no storing will be necessary.

File: discretize/MeshIO.py (hunk 2 above)
Comment (thast) [reply discretize/MeshIO.py]: we can actually make this function much simpler. Will do.

File: discretize/MeshIO.py [hunk 5]
```diff
@@ -118,7 +118,7 @@ def unpackdx(fid, nrows):
         return tensMsh
 
     @classmethod
-    def readUBC(TensorMesh, fileName, meshdim=None):
+    def readUBC(TensorMesh, fileName, meshdim=None, folder=''):
```
Comment (lheagy) [inline discretize/MeshIO.py]: do we want `folder` or `directory` ? 
cc @rowanc1: do you know what is more common?

Comment (thast) [reply discretize/MeshIO.py]: Indeed directory does sound better.

File: discretize/MeshIO.py [hunk 6]
```diff
@@ -118,51 +124,60 @@ def unpackdx(fid, nrows):
...
         if meshdim == None:
             # Read the file as line strings, remove lines with comment = !
-            msh = np.genfromtxt(fileName, delimiter='\n', dtype=np.str, comments='!', max_rows=1)
+            msh = np.genfromtxt(fname, delimiter='\n', dtype=np.str, comments='!', max_rows=1)
             # Fist line is the size of the model
             sizeM = np.array(msh.ravel()[0].split(), dtype=float)
             # Check if the mesh is a UBC 2D mesh
             if sizeM.shape[0] == 1:
-                Tnsmsh = TensorMesh._readUBC_2DMesh(fileName)
+                Tnsmsh = TensorMesh._readUBC_2DMesh(fname)
```
Comment (lheagy) [inline discretize/MeshIO.py]: to simplify this a bit (and reduce redundancy), why don't we just assign `meshdim = 2` here or `meshdim = 3` below, and then  check meshdim and call the appropriate `readUBC` method after that. 

Thoughts?

Comment (thast) [reply discretize/MeshIO.py]: I agree, this is a function I wrote in a previous PR but it was a redundant check between what the user expects and what the machine read.

**Comment by thast** [comment]
Thanks @lheagy, can you merge it for me please? Does not seem I have the rights to do so 
("Only those with write access to this repository can merge pull requests.")

**Pull request merged**